# ensegment: default program

In [2]:
from ensegment import *

## Documentation

This case shows the default program's result. 

In [2]:
Pw = Pdist(data=datafile("../data/count_1w.txt"))
segmenter = Segment(Pw)
with open("../data/input/dev.txt") as f:
    for line in f:
        print(" ".join(segmenter.segment(line.strip())))

choose spain
this is a test
who represents
experts exchange
speed of art
unclimatechangebody
we are the people
mentionyourfaves
now playing
the walking dead
follow me
we are the people
mentionyourfaves
check domain
big rock
name cheap
apple domains
honesty hour
being human
follow back
social media
30secondstoearth
current ratesoughttogodown
this is insane
what is my name
is it time
let us go
me too
nowthatcherisdead
advice for young journalists


## Analysis

The F-score of default program for dev.txt input was 0.82. Group members wondered what scores the wrong sentences got, and how much it was higher than the correct sentences.

# ensegment: top5 program

## Documentation

This case shows the top 5 of scores of the ensegment results. The scores are logged to be mroe readable. "unclimatechangebody", "gerger, and "gergerger" are used for the test.

In [1]:
from temp_test import *

['unclimatechangebody']
-11.76946906492656
['unclimatechangebod', 'y']
-15.248877454922333
['u', 'nclimatechangebody']
-15.284594107909584
['unclimatechange', 'body']
-15.40361194398702
['un', 'climate', 'change', 'body']
-15.532385601337921
['unclimatechangebody']


In [1]:
from temp_test import *

['ger', 'ger']
-10.52313386778198
['gerger']
-11.76946906492656
['gerg', 'er']
-11.780843268791939
['g', 'er', 'ger']
-13.207159111345849
['ge', 'r', 'ger']
-13.223195726461629
['ger', 'ger']


In [1]:
from temp_test import *

['gergerger']
-11.76946906492656
['gergerge', 'r']
-15.029017865766795
['g', 'ergerger']
-15.181438477228347
['ger', 'ger', 'ger']
-15.784700801672969
['gergerg', 'er']
-16.30309183007963
['gergerger']


## Analysis

We found that the unseen 1 word sentence has the constant value of missingfn: -11.76946906492656(1/N) from the result, and this was higher than the correct result for some sentences. The one observaition is that when the correct result has small segements, the result was correct, but when the segements are many, the scores go below the constant score of missingfn because program multiplies P(w)s the number of segements times for the score.

# ensegment: testcase1 program

## Documentation

This case shows the result of the first test case. We set the missingfn value to 0 as the minimum score.

change: else: return self.missingfn(key, self.N)

to: else: return 0

In [3]:
Pw = Pdist(data=datafile("../data/count_1w.txt"))
segmenter = Segment(Pw)
with open("../data/input/dev.txt") as f:
    for line in f:
        print(" ".join(segmenter.segment(line.strip())))

choose spain
this is a test
who represents
experts exchange
speed of art
un climate change body
we are the people
mention your faves
now playing
the walking dead
follow me
we are the people
mention your faves
check domain
big rock
name cheap
apple domains
honesty hour
being human
follow back
social media
3 0 seconds to earth
current rate sought to go down
this is insane
what is my name
is it time
let us go
me too
now thatcher is dead
advice for young journalists


In [4]:
Pw = Pdist(data=datafile("../data/count_1w.txt"))
segmenter = Segment(Pw)
with open("../data/input/test.txt") as f:
    for line in f:
        print(" ".join(segmenter.segment(line.strip())))

h o w t o b r e a k u p i n 5 words
what makes god smile
1 0 people who mean alot to me
w o r s t d a y i n 4 words
l o v e s t o r y i n 5 words
t o p 3 favourite comics
1 0 breakup lines
things that make you smile
best female athlete
w o r s t b o s s i n 5 words
now is the time for all good
it is a truth universally acknowledged
when in the course of human events it becomes necessary
it was a bright cold day in april and the clocks were striking thirteen
it was the best of times it was the worst of times it was the age of wisdom it was the age of foolishness
as gregor samsa awoke one morning from uneasy dreams he found himself transformed in his bed into a gigantic insect
in a hole in the ground there lived a hobbit not a nasty dirty wet hole filled with the ends of worms and an oozy smell nor yet a dry bare sandy hole with nothing in it to sitdown on or to eat it was a hobbit hole and that means comfort
far out in the uncharted backwaters of the unfashionable end of the western spi

## Analysis

The F-score of dev.output was 0.98, but the result of test.output looked messy. We found that if missingfn value is 0, then ensegment program does not work properly when it meets unseen words. So we decided to give missingfn non-zero probability.

# ensegment: testcase2 program

## Documentation

This case shows the result of the second test case. We set the missingfn value to 1/(N*N) as the minimum score.

change: self.missingfn = missingfn or (lambda k, N: 1./(N))

to: self.missingfn = missingfn or (lambda k, N: 1./(N*N))

In [4]:
Pw = Pdist(data=datafile("../data/count_1w.txt"))
segmenter = Segment(Pw)
with open("../data/input/dev.txt") as f:
    for line in f:
        print(" ".join(segmenter.segment(line.strip())))

choose spain
this is a test
who represents
experts exchange
speed of art
un climate change body
we are the people
mention your faves
now playing
the walking dead
follow me
we are the people
mention your faves
check domain
big rock
name cheap
apple domains
honesty hour
being human
follow back
social media
30secondstoearth
current rate sought to go down
this is insane
what is my name
is it time
let us go
me too
now thatcher is dead
advice for young journalists


In [5]:
Pw = Pdist(data=datafile("../data/count_1w.txt"))
segmenter = Segment(Pw)
with open("../data/input/test.txt") as f:
    for line in f:
        print(" ".join(segmenter.segment(line.strip())))

howtobreakupin5words
what makes god smile
10peoplewhomeanalot to me
worstdayin4words
lovestoryin5words
top3favouritecomics
10breakuplines
things that make you smile
best female athlete
worstbossin5words
now is the time for all good
it is a truth universally acknowledged
when in the course of human events it becomes necessary
it was a bright cold day in april and the clocks were striking thirteen
it was the best of times it was the worst of times it was the age of wisdom it was the age of foolishness
as gregor samsa awoke one morning from uneasy dreams he found himself transformed in his bed into a gigantic insect
in a hole in the ground there lived a hobbit not a nasty dirty wet hole filled with the ends of worms and an oozysmellnoryetadryb are sandy hole with nothing in it to sitdown on or to eat it was a hobbit hole and that means comfort
far out in the uncharted backwaters of the unfashionable end of the western spiral arm of the galaxy lies a small un regarded yellow sun


## Analysis

The result of F-score was 0.97, and we realised that the reducing missingfn to some constant value does not solve the core problem. This problem always happend when the long words is unseen; therefore, we applied cost for missingfn according to the its word length.

# ensegment: testcase3 program

## Documentation

This case shows the result of the third test case. We set the missingfn value to 1/(N**len(k)) as the minimum score.

change: self.missingfn = missingfn or (lambda k, N: 1./(N))

to: self.missingfn = missingfn or (lambda k, N: 1./(N**len(k)))

In [3]:
Pw = Pdist(data=datafile("../data/count_1w.txt"))
segmenter = Segment(Pw)
with open("../data/input/dev.txt") as f:
    for line in f:
        print(" ".join(segmenter.segment(line.strip())))

choose spain
this is a test
who represents
experts exchange
speed of art
un climate change body
we are the people
mention your faves
now playing
the walking dead
follow me
we are the people
mention your faves
check domain
big rock
name cheap
apple domains
honesty hour
being human
follow back
social media
3 0 seconds to earth
current rate sought to go down
this is insane
what is my name
is it time
let us go
me too
now thatcher is dead
advice for young journalists


In [4]:
Pw = Pdist(data=datafile("../data/count_1w.txt"))
segmenter = Segment(Pw)
with open("../data/input/test.txt") as f:
    for line in f:
        print(" ".join(segmenter.segment(line.strip())))

how to breakup in 5 words
what makes god smile
1 0 people who mean alot to me
worst day in 4 words
love story in 5 words
top 3 favourite comics
1 0 breakup lines
things that make you smile
best female athlete
worst boss in 5 words
now is the time for all good
it is a truth universally acknowledged
when in the course of human events it becomes necessary
it was a bright cold day in april and the clocks were striking thirteen
it was the best of times it was the worst of times it was the age of wisdom it was the age of foolishness
as gregor samsa awoke one morning from uneasy dreams he found himself transformed in his bed into a gigantic insect
in a hole in the ground there lived a hobbit not a nasty dirty wet hole filled with the ends of worms and an oozy smell nor yet a dry bare sandy hole with nothing in it to sitdown on or to eat it was a hobbit hole and that means comfort
far out in the uncharted backwaters of the unfashionable end of the western spiral arm of the galaxy lies a small 

## Analysis

We could see the great improvement from the test.output; however, there was one more problem which is that integer is considered as unseen words in the count_1w.text. We tried numbers to be together in the sentence.

# ensegment: final testcase program

## Documentation

This case shows the result of the final test case. We modified splits function not to split numbers. F-score for dev.out was 1.0, and test.out looks great except for the "un regarded"(unregarded).

change: return [(text[:i+1], text[i+1:]) for i in range(min(len(text), L))]

In [2]:
Pw = Pdist(data=datafile("../data/count_1w.txt"))
segmenter = Segment(Pw)
with open("../data/input/dev.txt") as f:
    for line in f:
        print(" ".join(segmenter.segment(line.strip())))

choose spain
this is a test
who represents
experts exchange
speed of art
un climate change body
we are the people
mention your faves
now playing
the walking dead
follow me
we are the people
mention your faves
check domain
big rock
name cheap
apple domains
honesty hour
being human
follow back
social media
30 seconds to earth
current rate sought to go down
this is insane
what is my name
is it time
let us go
me too
now thatcher is dead
advice for young journalists


In [3]:
Pw = Pdist(data=datafile("../data/count_1w.txt"))
segmenter = Segment(Pw)
with open("../data/input/test.txt") as f:
    for line in f:
        print(" ".join(segmenter.segment(line.strip())))

how to breakup in 5 words
what makes god smile
10 people who mean alot to me
worst day in 4 words
love story in 5 words
top 3 favourite comics
10 breakup lines
things that make you smile
best female athlete
worst boss in 5 words
now is the time for all good
it is a truth universally acknowledged
when in the course of human events it becomes necessary
it was a bright cold day in april and the clocks were striking thirteen
it was the best of times it was the worst of times it was the age of wisdom it was the age of foolishness
as gregor samsa awoke one morning from uneasy dreams he found himself transformed in his bed into a gigantic insect
in a hole in the ground there lived a hobbit not a nasty dirty wet hole filled with the ends of worms and an oozy smell nor yet a dry bare sandy hole with nothing in it to sitdown on or to eat it was a hobbit hole and that means comfort
far out in the uncharted backwaters of the unfashionable end of the western spiral arm of the galaxy lies a small un